packages/shared/src/components/inputs/Input.svelte

```diff
@@ -103,11 +103,8 @@
             if (pasteVal?.indexOf('e') >= 0 || pasteVal?.indexOf('-') >= 0) {
                 event.preventDefault()
             } else if (float) {
-                const val = parseCurrency(pasteVal)
-                // Discard any numbers we can't parse as floats
-                if (Number.isNaN(val)) {
```

Comment:
This check is already happening is `parseCurrency`. Although, you're comment did unravel a mistake in the parseCurrency function.

Instead of calling `preventDefault`, parseCurrency returns '0' in case the value cannot be parsed as a float.